File: tmp_mobile_test.py
from playwright.sync_api import sync_playwright
import json
with sync_playwright() as p:
    browser = p.chromium.launch()
    iphone = p.devices['iPhone 14']
    context = browser.new_context(**iphone)
    page = context.new_page()
    page.goto('http://127.0.0.1:3000/login')
    page.screenshot(path='C:/Users/princ/OneDrive/Desktop/swapSpot/login-mobile.png', full_page=True)
    body_size = page.evaluate('''() => ({ w: document.body.clientWidth, h: document.body.clientHeight, scrollH: document.body.scrollHeight, htmlH: document.documentElement.clientHeight, htmlScrollH: document.documentElement.scrollHeight })''')
    print(json.dumps(body_size))
    browser.close()

File: tmp_mobile_test.js
const { chromium, devices } = require('playwright');
(async () => {
  const browser = await chromium.launch();
  const iphone = devices['iPhone 14'];
  const context = await browser.newContext(iphone);
  const page = await context.newPage();
  await page.goto('http://127.0.0.1:3000/notifications');
  await page.screenshot({ path: 'C:/Users/princ/OneDrive/Desktop/swapSpot/notifications-mobile.png', fullPage: true });
  const bodySize = await page.evaluate(() => ({ w: document.body.clientWidth, h: document.body.clientHeight, scrollH: document.body.scrollHeight, htmlH: document.documentElement.clientHeight, htmlScrollH: document.documentElement.scrollHeight }));
  console.log(JSON.stringify(bodySize));
  await browser.close();
})();
